app/services/code_climate/projects_summary_service.rb

```diff
@@ -58,16 +68,38 @@ def metrics_in_given_technologies
 
     def metrics_in_time_period
       if from.positive?
-        metrics_in_department.where(snapshot_time: from.weeks.ago..Time.zone.now)
+        metrics_in_department.with_activity_after(from.weeks.ago)
       else
         metrics_in_department
       end
     end
 
     def metrics_in_department
-      CodeClimateProjectMetric
-        .joins(project: { language: :department })
-        .where(departments: { id: department.id })
+      Project.joins(:code_climate_project_metric)
+             .joins(:language)
+             .where(language: department.languages)
+             .distinct
+             .relevant
+    end
+
+    def projects_without_cc_count
+      (ids_without_rate - ids_with_rate).count
+    end
+
+    def ids_without_rate
+      projects_without_cc_count = Project.with_no_cc_nor_cc_rate
```

Comment:
names look a bit confusing, you are using the same name for this variable as for the method above, but both yield different results, right?

Reply:
my bad, this code was formerly on the method above, will change the names, thanks